provider/bidengine/order_test.go

```diff
@@ -0,0 +1,591 @@
+package bidengine
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/enigmampc/btcutil/bech32"
+	"github.com/ovrclk/akash/sdkutil"
+	"google.golang.org/grpc"
+	"math/rand"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ovrclk/akash/provider/cluster"
+	"github.com/ovrclk/akash/provider/session"
+	"github.com/ovrclk/akash/pubsub"
+	atypes "github.com/ovrclk/akash/types"
+	dtypes "github.com/ovrclk/akash/x/deployment/types"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+	ptypes "github.com/ovrclk/akash/x/provider/types"
+
+	"github.com/ovrclk/akash/client"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+var errNotImplemented = errors.New("not implemented")
+
+type fauxQueryClient int
+
+func (_ fauxQueryClient) Providers(ctx context.Context, in *ptypes.QueryProvidersRequest, opts ...grpc.CallOption) (*ptypes.QueryProvidersResponse, error) {
+	return nil, errNotImplemented
+}
+
+func (_ fauxQueryClient) Provider(ctx context.Context, in *ptypes.QueryProviderRequest, opts ...grpc.CallOption) (*ptypes.QueryProviderResponse, error) {
+	return nil, errNotImplemented
+}
+
+// Deployments queries deployments
+func (_ fauxQueryClient) Deployments(ctx context.Context, in *dtypes.QueryDeploymentsRequest, opts ...grpc.CallOption) (*dtypes.QueryDeploymentsResponse, error) {
+	return nil, errNotImplemented
+}
```

Comment:
If we don't have mocks for this, let's generate one.  See the pattern [here](https://github.com/ovrclk/akash/blob/494572ac32a7a48d801a27397e6d2047d80419ec/make/codegen.mk#L5-L12)